src/metabase/driver/sql/query_processor.clj

```diff
@@ -209,6 +210,16 @@
   (fn [driver seconds-or-milliseconds _] [(driver/dispatch-on-initialized-driver driver) seconds-or-milliseconds])
   :hierarchy #'driver/hierarchy)
 
+(defmulti cast-temporal-string
```

Comment:
I think it's reasonable to have these here. Non-SQL as they are in a lot of ways an artefact of SQL semantics. If we ever come to a situation that we want to support this for other drivers, we can always move it to the main driver namespace, but for now I think smaller surface area wins.

Reply:
thanks @sbelak . It felt right and kinda mimiced what the unix timestamp was doing but was worried i was foreclosing other databases unnecessarily.

Reply:
Well done! 🏆